File: banchus-common/src/main/java/pe/nanamochi/banchus/database/repository/SessionRepository.java
package pe.nanamochi.banchus.database.repository;

import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import pe.nanamochi.banchus.database.entity.Session;

@Repository
public interface SessionRepository extends JpaRepository<Session, UUID> {
  Optional<Session> findByUser_IdAndPrimarySessionTrue(int userId);

  Optional<Session> findByUser_UsernameIgnoreCaseAndPrimarySessionTrue(String username);
}

File: banchus-common/src/main/kotlin/pe/nanamochi/banchus/database/repository/SessionRepository.kt
package pe.nanamochi.banchus.database.repository

import java.util.UUID
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pe.nanamochi.banchus.database.entity.Session

@Repository
interface SessionRepository : JpaRepository<Session, UUID> {
    fun findSessionById(id: UUID): Session?

    fun findByUserIdAndPrimarySessionTrue(userId: Int): Session?

    fun findByUserUsernameIgnoreCaseAndPrimarySessionTrue(username: String): Session?
}
